### PR TITLE
HR-1205 -> master

### PR DESCRIPTION
**Pull request by darrenjaworski**
FindTransactions builder...

### CODE REVIEW

File: src/Gateways/PorticoConnector.ts [hunk 1]
```diff
@@ -1324,4 +1337,13 @@ export class PorticoConnector extends XmlGateway implements IPaymentGateway {
 
     return result;
   }
+
+  protected hydrateFoundTransactions(root: Element): Transaction {
```
Comment (darrenjaworski) [inline src/Gateways/PorticoConnector.ts:1341]: @wdesimini which fields do we need out of this request? Here are all available: https://cert.api2.heartlandportico.com/Gateway/PorticoSOAPSchema/build/Default/webframe.html#Portico_xsd~e-PosResponse~e-Ver1.0~e-Transaction~e-FindTransactions.html

@lundhjem

Comment (wdesimini) [reply src/Gateways/PorticoConnector.ts:1341]: I think we're mainly looking for the transaction amount (looks like either Amt or AuthAmt) to stick into the amount parameter of the reversal request, but maybe just return the whole transaction?

@lundhjem could you confirm/provide more info?

File: src/Gateways/PorticoConnector.ts [hunk 2]
```diff
@@ -989,6 +999,7 @@ export class PorticoConnector extends XmlGateway implements IPaymentGateway {
     builder: ReportBuilder<T>,
   ): T {
     // todo: handle non-200 responses
+    debugger;
```
Comment (darrenjaworski) [inline src/Gateways/PorticoConnector.ts:1002]: be sure to remove this

File: src/Gateways/PorticoConnector.ts (hunk 1 above)
Comment (darrenjaworski) [reply src/Gateways/PorticoConnector.ts:1341]: here is all the data these are returning: 

Screen Shot 2020-06-05 at 10 37 00 AM
Screen Shot 2020-06-05 at 10 36 53 AM

File: src/Entities/Transaction.ts [hunk 3]
```diff
@@ -58,6 +61,18 @@ export class Transaction {
     return transaction;
   }
 
+  // TODO see if we actually need this
+  public static fromClientId(
+    clientTransactionId: string,
+    paymentMethodType = PaymentMethodType.Credit,
+  ) {
```
Comment (killrazor) [inline src/Entities/Transaction.ts:68]: Consider adding the return type so that it doesn't have to be inferred

File: tsconfig.json [hunk 4]
```diff
@@ -22,7 +22,6 @@
   ],
   "exclude": [
     "node_modules",
-    "lib",
-    "netsuite"
```
Comment (killrazor) [inline tsconfig.json:26]: Thank you!

File: src/Entities/Transaction.ts (hunk 3 above)
Comment (darrenjaworski) [reply src/Entities/Transaction.ts:68]: I don't think we're going to need this block of code, i forgot to remove it.

File: src/Gateways/PorticoConnector.ts (hunk 1 above)
Comment (lundhjem) [reply src/Gateways/PorticoConnector.ts:1341]: @darrenjaworski Noice!! Thanks for sharing

**Pull request merged**